### ISSUE TITLE
False positive for no-incompatible-property-type with converter passed in

### ISSUE THREAD

**Issue by bgotink** [runem/lit-analyzer#349]
I've got a factory function creating attribute converters, but when passing these into `@property` they don't seem to be picked up by the `no-incompatible-property-type` check

```ts
// causes warning
@property({converter: convert(Iban)})
declare warning: Iban;

// doesn't cause warning
@property({converter: {...convert(Iban)}})
declare noWarning: Iban;

// doesn't cause warning either
@property({converter: ibanConverter}) // with const ibanConverter = convert(Iban) 
declare alsoNoWarning: Iban;
```
